engine/chunk/src/main/java/io/deephaven/chunk/util/pools/MultiChunkPool.java

```diff
@@ -4,17 +4,50 @@
 package io.deephaven.chunk.util.pools;
 
 import io.deephaven.chunk.ChunkType;
+import io.deephaven.chunk.ResettableBooleanChunk;
+import io.deephaven.chunk.ResettableByteChunk;
+import io.deephaven.chunk.ResettableCharChunk;
+import io.deephaven.chunk.ResettableDoubleChunk;
+import io.deephaven.chunk.ResettableFloatChunk;
+import io.deephaven.chunk.ResettableIntChunk;
+import io.deephaven.chunk.ResettableLongChunk;
+import io.deephaven.chunk.ResettableObjectChunk;
+import io.deephaven.chunk.ResettableShortChunk;
+import io.deephaven.chunk.ResettableWritableBooleanChunk;
+import io.deephaven.chunk.ResettableWritableByteChunk;
+import io.deephaven.chunk.ResettableWritableCharChunk;
+import io.deephaven.chunk.ResettableWritableDoubleChunk;
+import io.deephaven.chunk.ResettableWritableFloatChunk;
+import io.deephaven.chunk.ResettableWritableIntChunk;
+import io.deephaven.chunk.ResettableWritableLongChunk;
+import io.deephaven.chunk.ResettableWritableObjectChunk;
+import io.deephaven.chunk.ResettableWritableShortChunk;
+import io.deephaven.chunk.WritableBooleanChunk;
+import io.deephaven.chunk.WritableByteChunk;
+import io.deephaven.chunk.WritableCharChunk;
+import io.deephaven.chunk.WritableDoubleChunk;
+import io.deephaven.chunk.WritableFloatChunk;
+import io.deephaven.chunk.WritableIntChunk;
+import io.deephaven.chunk.WritableLongChunk;
+import io.deephaven.chunk.WritableObjectChunk;
+import io.deephaven.chunk.WritableShortChunk;
+import io.deephaven.chunk.attributes.Any;
 import org.jetbrains.annotations.NotNull;
 
 import java.util.Collections;
 import java.util.EnumMap;
 import java.util.Map;
+import java.util.function.Supplier;
 
 /**
  * Provides a set of per-type {@link ChunkPool}s. Normally accessed via a {@link ThreadLocal}, to allow some threads to
  * share a common pool and others to allocate their own.
+ * <p>
+ * This type isn't compatible with GWT, and needs to be replaced with a different class that doesn't use the soft pool
+ * instances.
```

Comment:
For some reason, I'm not in love with having comments about GWT around the rest of the code base. This one, in particular, seems more like a note to you describing future work than anything for readers of `MultiChunkPool` to care about.

Reply:
Fair - we're going to need to address the topic somewhere though so that other developers can:
 * see that their changes affect the js api, either by breaking things accidentally, or by automatically adding features there
 * see that in some cases that isn't true, where the js api has to maintain its own separate copy of classes

It could all live in web/client-api's developer docs (i.e. readme and such), but I'm not sure that will be enough. Unless we get very aggressive about splitting various utility types, we're inevitably going to need to add actual `@GwtIncompatible` annotations here and there in source too.